### PR TITLE
Algorithm modification according to new result structure + path added in result structure

### PR DESCRIPTION
**Pull request by gerardcel**

### CODE REVIEW

File: src/algorithm.jl [hunk 1]
```diff
@@ -22,14 +22,17 @@ Run the algorithm `algo` on the stochastic program `sp` until the termination cr
...
-    while (paths === nothing || getstatus(paths[1]) != :Infeasible) && !stop(stopcrit, info)
-        @timeit info.timer "iteration $(niterations(info)+1)" paths, result = iterate!(sp, algo, info.timer, verbose)
+    while !stop(stopcrit, info)
+        @timeit info.timer "iteration $(niterations(info)+1)" result = iterate!(sp, algo, info.timer, verbose)
+        paths = result.paths
         push!(info.results, result)
         if verbose >= 3
             print_iteration_summary(info)
         end
+        if getstatus(paths[1]) != :Infeasible
```
Comment (blegat) [inline src/algorithm.jl]: You can do `getstatus(result.paths[1].sol_scenario[1])` here directly and not create the `paths` variable.

File: src/algorithm.jl [hunk 2]
```diff
@@ -76,20 +79,18 @@ function forward_pass(sp::AbstractStochasticProgram, algo::AbstractAlgorithm, to
...
-
-    paths, forward_stats
+    result.paths = paths
+    result.upperbound = z_UB
+    result.σ_UB = σ
 end
 
 """
-    backward_pass!(sp::AbstractStochasticProgram, algo::AbstractAlgorithm, paths::Vector{Path}, to::TimerOutput, verbose)
+    backward_pass!(sp::AbstractStochasticProgram, algo::AbstractAlgorithm, to::TimerOutput, result::Result, verbose)
```
Comment (frapac) [inline src/algorithm.jl]: 80C!

File: src/algorithm.jl [hunk 3]
```diff
@@ -76,20 +79,18 @@ function forward_pass(sp::AbstractStochasticProgram, algo::AbstractAlgorithm, to
...
 
-Run the backward pass of algorithm `algo` on the stochastic program `sp` for the paths `paths` with verbose level `verbose`.
+Run the backward pass of algorithm `algo` on the stochastic program `sp` with verbose level `verbose` and use paths and
```
Comment (frapac) [inline src/algorithm.jl]: 80C!

File: src/algorithm.jl [hunk 4]
```diff
@@ -117,10 +118,10 @@ generated during the forward pass and σ represents the standard deviation of th
 function compute_bounds end
 
 """
-    process(sp::AbstractStochasticProgram, algo::AbstractAlgorithm, paths::Vector{Path}, to::TimerOutput)
+    process(sp::AbstractStochasticProgram, algo::AbstractAlgorithm, to::TimerOutput, result::Result, verbose)
```
Comment (frapac) [inline src/algorithm.jl]: 80C!

File: src/algorithm.jl [hunk 5]
```diff
@@ -117,10 +118,10 @@ generated during the forward pass and σ represents the standard deviation of th
...
 
 Function called at the end of function iterate! to update necessary elements related with the implementation of the
 algorithm `algo` on the stochastic program `sp` considering forward paths `paths`.
 """
-function process!(sp::AbstractStochasticProgram, algo::AbstractAlgorithm, paths::Vector{Path}, to::TimerOutput, verbose)
+function process!(sp::AbstractStochasticProgram, algo::AbstractAlgorithm, to::TimerOutput, result::Result, verbose)
```
Comment (frapac) [inline src/algorithm.jl]: 80C!

**Review by frapac — APPROVED** [review]
Ok to merge!

**Review by blegat — APPROVED** [review]
Ok once CI passes :)

**Pull request merged**